test/app-luatest/gh_9806_http_client_version_test.lua

```diff
@@ -0,0 +1,73 @@
+local t = require('luatest')
+local treegen = require('test.treegen')
+local justrun = require('test.justrun')
+local socket = require('socket')
+
+local g = t.group("versions", {
+    --'GET / HTTP/1.1'
+    { req = 'http', version = nil, res = '1.1' },
+    --'GET / HTTP/1.1'
+    { req = 'http', version = '1.1', res = '1.1' },
+    --Correct line
+    { req = 'http', version = '2',
+      res = 'Connection: Upgrade, HTTP2-Settings' },
+    --'GET / HTTP/1.1'
+    { req = 'http', version = '2-tls', res = '1.1' },
+    --'[HTTP/2] [1] OPENED stream'
+    { req = 'http', version = '2-prior-knowledge', res = '[1] OPENED stream' },
+    --'ALPN: curl offers h2,http/1.1'
+    { req = 'https', version = nil, res = 'ALPN: curl offers h2,http' },
+    --'ALPN: curl offers http/1.1'
+    { req = 'https', version = '1.1', res = 'ALPN: curl offers http' },
+    --'ALPN: curl offers h2,http/1.1'
+    { req = 'https', version = '2', res = 'ALPN: curl offers h2,http' },
+    --'ALPN: curl offers h2,http/1.1'
+    { req = 'https', version = '2-tls', res = 'ALPN: curl offers h2,http' },
+    --'ALPN: curl offers h2,http/1.1'
+    { req = 'https', version = '2-prior-knowledge',
+      res = 'ALPN: curl offers h2,http' },
```

Comment:
Please add whitespace after dashes and dots at the end of comments.

Reply:
These comments were temporary, I've already deleted them